src/neo4j/mod.rs

```diff
@@ -0,0 +1,702 @@
+    #[must_use]
+    pub fn with_user(mut self, user: impl Into<String>) -> Self {
+        self.user = Value::Value(user.into());
+        self
+    }
+
+    /// Set the password to use.
+    #[must_use]
+    pub fn with_password(mut self, pass: impl Into<String>) -> Self {
+        self.pass = Value::Value(pass.into());
+        self
+    }
+
+    /// Do not use any authentication on the testcontainer.
+    ///
+    /// Setting this will override any prior usages of [`Self::with_user`] and
+    /// [`Self::with_password`].
+    pub fn without_authentication(mut self) -> Self {
+        self.user = Value::Unset;
+        self.pass = Value::Unset;
+        self
+    }
+
+    /// Use the enterprise edition of Neo4j.
+    ///
+    /// # Note
+    /// Please have a look at the [Neo4j Licensing page](https://neo4j.com/licensing/).
+    /// While the Neo4j Community Edition can be used for free in your projects under the GPL v3 license,
+    /// Neo4j Enterprise edition needs either a commercial, education or evaluation license.
+    pub fn with_enterprise_edition(
+        mut self,
+    ) -> Result<Self, Box<dyn std::error::Error + Sync + Send + 'static>> {
+        const ACCEPTANCE_FILE_NAME: &str = "container-license-acceptance.txt";
+
+        let version = Self::value(&self.version).expect("Version is always set");
+        let image = format!("neo4j:{}-enterprise", version);
+
+        let acceptance_file = std::env::current_dir()
+            .ok()
+            .map(|o| o.join(ACCEPTANCE_FILE_NAME));
+
+        let has_license_acceptance = acceptance_file
+            .as_deref()
+            .and_then(|o| std::fs::File::open(o).ok())
+            .into_iter()
+            .flat_map(|o| std::io::BufReader::new(o).lines())
+            .any(|o| o.map_or(false, |line| line.trim() == image));
+
+        if !has_license_acceptance {
+            return Err(format!(
+                concat!(
+                    "You need to accept the Neo4j Enterprise Edition license by ",
+                    "creating the file `{}` with the following content:\n\n\t{}",
+                ),
+                acceptance_file.map_or_else(
+                    || ACCEPTANCE_FILE_NAME.to_owned(),
+                    |o| { o.display().to_string() }
+                ),
+                image
+            )
+            .into());
+        }
+
+        self.enterprise = true;
+        Ok(self)
+    }
+
+    /// Add Neo4j lab plugins to get started with the database.
+    #[must_use]
+    pub fn with_neo4j_labs_plugin(mut self, plugins: &[Neo4jLabsPlugin]) -> Self {
+        self.plugins.extend_from_slice(plugins);
+        self
+    }
+
+    /// Create a new instance of a Neo4j image of the given version with the default user and password.
+    ///
+    /// # Panics
+    ///
+    /// If the version is not valid according to the format described in [`Self::with_version()`].
+    #[deprecated(since = "0.2.0", note = "Use `from_env().with_version()` instead.")]
```

Comment:
Why do you want to include immediate deprecated functions? Is it for downwards compatibility for projects this image is already in use?

Reply:
I believe all deprecated functions should be deleted, also not clear what does `0.2.0` mean here. Likely the version of another crate?

Reply:
I guess 0.2.0 would the next version of this crate considering merging this PR would include a new feature.

Reply:
Ah my bad. I have the module as a separate crate in https://github.com/knutwalker/neo4j-testcontainers-rs and copied it from there. The deprecation note and versions refer to that crate. I'll remove the deprecated methods from this PR.